In [ ]:
import numpy as np
from scipy import stats

### **Bài 1.** Trên tập mẫu gồm 100 số liệu, người ta tính được $\bar{x} = 0.1$ và $s = 0.014$. Xác định gia tri trung binh voi khoảng tin cậy 95% cho giá trị trung bình thật.

Gọi $\mu$ là giá trị trung bình thật của tập mẫu. Ta có, $\bar{x}$ là giá trị trung bình của 100 số liệu.

Độ tin cậy $95\% = 1 - \alpha$ => $\alpha=0.05$

Ta có $n = 100 > 30$. => áp dụng định lý giới hạn trung tâm, ta có:
$$\bigg(\bar{x} - z_{\frac{\alpha}{2}}\frac{s}{\sqrt{n}},\bar{x}+z_{\frac{\alpha}{2}}\frac{s}{\sqrt{n}}\bigg)$$
Trong đó, $\alpha = 5\% = 0.05$, $z_{1-\frac{\alpha}{2}}=z_{0.975}=1.96$, $\bar{x}=0.1$, $s=0.014$


In [ ]:
x = 0.1
s = 0.014
n = 100
alpha = 0.05
z = stats.norm.ppf(1-alpha/2)
print(x-z*s/np.sqrt(n), x+z*s/np.sqrt(n))

0.09725605042164392 0.10274394957835609


Vậy khoảng ước lượng 95% cho giá trị trung bình thật là: $(0.097256,0.102744)$

### **Bài 2** Quan sát chiều cao $X$ (cm) của một số nữ giới, ta ghi nhận số liệu như sau:
| x(cm)    | 140-145 | 145-150 | 150-155 | 155 − 160 | 160 − 165 | 165 − 170
| -------- | ------- | ------- | ------- | ------- | ------- | ------- |
| Số người  | 1 | 3 | 7|  9| 5| 2

a) Tính $\bar{x}$ và $s^2$
Ta viết lại bảng tần số:

| x(cm)    | 142.5 | 147.5 | 152.5 | 157.5 | 162.5 | 167.5
| -------- | ------- | ------- | ------- | ------- | ------- | ------- |
| Số người  | 1 | 3 | 7|  9| 5| 2

Ta có: $$n=1+3+7+9+5+2=27$$
$$\bar{x}=\frac{142.5\times1+147.5\times3+152.5\times7+157.5\times9+162.5\times5+167.5\times2}{27}=156.2$$
$$s^2 = \frac{1\times(142.5-156.2)^2+3\times(147.5-156.2)^2+7\times(152.5-156.2)^2+9\times(157.5-156.2)^2+5\times(162.5-156.2)^2+2\times(167.5-156.2)^2}{27-1}=37.678$$


b) Ước lượng $\mu$ với độ tin cậy 95%

Ta có, $n=27<30$, Vì mẫu nhỏ và chưa biết $\sigma^2$ nên chúng ta sử dụng phân phối t với bậc tự do $n-1=26$ để ước lượng và độ tin cậy $95\% = 1 - \alpha$ => $\alpha=0.05$

Vậy ta có:
$$\bigg(\bar{x} - t^{n-1}_{\frac{\alpha}{2}}\frac{s}{\sqrt{n}},\bar{x}+t^{n-1}_{\frac{\alpha}{2}}\frac{s}{\sqrt{n}}\bigg)$$

In [ ]:
x=156.2
n = 27
alpha = 0.05
s = 37.678**0.5
t = stats.t.ppf(1-alpha/2, n-1)
print(x-t*s/np.sqrt(n), x+t*s/np.sqrt(n))

153.77179281522737 158.6282071847726


Vậy khoảng ước lượng 95% cho $\mu$ là: $(153.77,158.63)$

### **Bài 3.** Một loại thuốc mới đem điều trị cho 50 người bị bệnh B, kết quả có 40 người khỏi bệnh.

a) Ước lượng tỷ lệ khỏi bệnh p nếu dùng thuốc đó điều trị với độ tin cậy 99%.

Ta có $n=50$; tỉ lệ để khỏi bệnh p là: $\hat{p}=\frac{40}{50}=0.8$; Độ tin câỵ là: $99\%=1-\alpha => \alpha=0.01$

Khoảng tin cậy cho tỉ lệ p là:
$$\bigg(\hat{p} - z_{\frac{\alpha}{2}}\sqrt{\frac{\hat{p}(\hat{p})}{n}},\bar{x}+z_{\frac{\alpha}{2}}\sqrt{\frac{\hat{p}(\hat{p})}{n}}\bigg)$$

In [ ]:
n = 50
p = 40/50
alpha = 0.01
z = stats.norm.ppf(1-alpha/2)
print(p-z*np.sqrt(p*(1-p)/n), p+z*np.sqrt(p*(1-p)/n))

0.654289090582524 0.945710909417476


**vậy ước lượng tỉ lệ khỏi bệnh p nếu dùng thuốc đó điều trị với độ tin cậy 99% là: $(0.654,0.9457)$**

b) Nếu muốn sai số ước lượng không quá 0.02 ở độ tin cậy 0.95 thì phải quan sát ít nhất bao nhiêu trường hợp?

Ta có sai số của ước lượng là: $E = z_{1-\frac{\alpha}{2}}\sqrt{\frac{\hat{p}(1-\hat{p})}{n}}$

Với độ tin cậy 95% thì ta có $\alpha=0.05$. Để có $E<0.02$ ta giải hệ phương trình sau:
$$z_{1-\frac{\alpha}{2}}\sqrt{\frac{\hat{p}(\hat{p})}{n}} \leq 0.02$$

In [ ]:
alpha = 0.05
z = stats.norm.ppf(1-alpha/2)
(p*(1-p))/((0.02/z)**2)

1536.5835282776495

Vậy cần quan sát ít nhất là 1537 trường hợp để có sai số ước lượng ko quá 0.02

### **Bài 4**
Trong điều kiện chăn nuôi bình thường, lượng sữa trung bình của 1 con bò là 14 kg/ngày. Nghi ngờ điều kiện chăn nuôi kém đi làm cho lượng sữa giảm xuống, người ta điều tra ngẫu nhiên 25 con và tính được lượng sữa trung bình của 1 con trong 1 ngày là 12.5 và độ lệch chuẩn s = 2.5. Với mức ý nghĩa α = 0.05. Hãy kết luận điều nghi ngờ nói trên. Giả thiết lượng sữa bò là 1 biến ngẫu nhiên chuẩn.

- Gọi $\mu$ là lượng sữa trung bình của một con bò trong 1 ngày với điều kiện chăn nuôi bị kém đi.
Ta cần kiểm định giả thuyết:

$H_0 : \mu \geq 14$: Điều kiện chăn nuôi không ảnh hưởng đến chất lượng sữa

$H_1: \mu < 14$: Điều kiện kém thì chất lượng cũng kém

Ta có $n = 25 < 30$;  $\bar{x} = 12.5$; $s = 2.5$, và lượng sữa bò là 1 biến ngẫu nhiên chuẩn, chưa biết $\sigma^2$ nên ta dùng phân phối $t$ để tính giá trị thống kê và p-values:
$$t_{obs}=\frac{\bar{x}-\mu}{\frac{s}{\sqrt{n}}}$$

$H_1: \mu < \mu_0$ nên $p_{value} = P(T_{n-1}<t_{obs})$

In [ ]:
x=12.5
s=2.5
n=25
alpha = 0.05
t_s = (x-14)/(s/np.sqrt(n))
print("T obs:", t_s)
p = stats.t.cdf(-np.abs(t_s))
print("p values: ", p)
if p < alpha:
    print("Bác bỏ H 0")
else:
    print("Chấp nhận H 0")

T obs: -3.0
p values:  0.003102868308262372
Bác bỏ H 0


Vậy với mức ý nghĩa $\alpha=0.05$ thì điều kiện chăn nuôi kém làm cho chất lượng sữa giảm đi.

### **Bài 5** Tỷ lệ phế phẩm của một nhà máy trước đây là 5%. Năm nay nhà máy áp dụng một biện pháp kỹ thuật mới. Để nghiên cứu tác dụng của biện pháp kỹ thuật mới, người ta lấy một mẫu gồm 800 sản phẩm để kiểm tra và thấy có 24 phế phẩm.

#### a) Với α = 0.01. Hãy cho kết luận về biện pháp kỹ thuật mới này?

Ta cần kiểm định giả thuyết:

$H_0 : p = 0.05$: biện pháp kỹ thuật mới không làm thay đổi tỷ lệ phế phẩm.

$H_1: p \neq 0.05$: biện pháp kỹ thuật mới làm thay đổi tỷ lệ phế phẩm.

- Các quan sát mẫu là một mẫu ngẫu nhiên đơn giản, các lần kiểm tra sản phẩm là độc lập nên tỷ lệ nhận được sản phẩm phế phẩm là tuân theo phân phối nhị thức.
- Gọi $p_o = 5\%$ là tỷ lệ phế phẩm của nhà máy khi chưa áp dụng biện pháp kỹ thuật mới. Ta có $n=800$ và $\hat{p}=\frac{24}{800}$.

Ta có $n=800 > 30$, $\hat{p}=0.03$, $np=800\times0.03=24 \geq 5$ và $n(1-p)=800(1-0.03)=776 \geq 5$. Vậy phân phối nhị thức của tỷ lệ mẫu có thể xấp xỉ bằng phân phối chuẩn.
$$z=\frac{\hat{p}-p}{\sqrt{\frac{pq}{n}}}$$
$$p_{values} = P(Z>z)$$

In [ ]:
n = 800
p = 24/800
p_hat = 0.05
alpha = 0.01
z = n**0.5*(p-p_hat)/((p_hat*(1-p_hat))**0.5)
print(z)
p_value = 2*(stats.norm.cdf(-z if z > 0 else z))
print(p_value)
if p_value < alpha:
    print("Từ chối giả thuyết H0")
else:
    print("Chấp nhận giả thuyết H0")

-2.595542738092201
0.00944416568783263
Từ chối giả thuyết H0


Vì $p_{values} < \alpha$ nên chúng ta chấp nhận giả thuyết $H_1$, tức là với mức ý nghĩa $\alpha=0.01$ biện pháp kỹ thuật mới làm thay đổi tỷ lệ phế phẩm.

#### b) Nếu nhà máy báo cáo tỷ lệ phế phẩm sau khi áp dụng biện pháp kỹ thuật mới là 2% thì có chấp nhận được không? (α = 0.01).

Tương tự ta có:

$H_0 : p = 0.02$: Tỷ lệ khi áp dụng biện pháp kỹ thuật mới là 2%

$H_1: p \neq 0.02$: Tỷ lệ khi áp dụng biện pháp kỹ thuật mới khác 2%

Tương tự ta tính $z$ và $p_{values}$

In [ ]:
n = 800
p = 24/800
p_hat = 0.02
alpha = 0.01
z = n**0.5*(p-p_hat)/((p_hat*(1-p_hat))**0.5)
print(z)
p_value = 2*(stats.norm.cdf(-z if z > 0 else z))
print(p_value)
if p_value < alpha:
    print("Từ chối giả thuyết H0")
else:
    print("Chấp nhận giả thuyết H0")

2.020305089104421
0.04335175126086268
Chấp nhận giả thuyết H0


Vì $p_{values} > \alpha$ nên chúng ta chấp nhận giả thuyết $H_0$. Vậy với mức ý nghĩa $\alpha=0.01$ không có đủ bằng chứng mẫu để đảm bảo biện pháp kỹ thuật mới có $2\%$ tỷ lệ phế phẩm.